### ISSUE TITLE
Return value of Grav\Common\Page\Page::getCacheKey() must be of the type string, null returned

### ISSUE THREAD

**Issue by tranduyhung** [getgrav/grav#2741]
After updating to Grav 1.6.17, we get the error ```Return value of Grav\Common\Page\Page::getCacheKey() must be of the type string, null returned``` in our custom plugin. The problem also happens to other plugins as well. Actually the problem happens when Grav tries to get cached media of a non-existing page.

To reproduces you can test with a standard Grav release:

* Install Grav and Admin plugin
* Try to edit a non-existing page , eg ```http://localhost:3001/admin/pages/this-page-doesn-t-exist``` and you can see page form as usual
* Install SEO plugin
* Try to access the above URL again and you get the error

The error happens in system/src/Grav/Common/Media/Traits/MediaTrait.php:

    public function getMedia()
    {
        if ($this->media === null) {
            $cache = $this->getMediaCache();

            // Use cached media if possible.
            $cacheKey = md5('media' . $this->getCacheKey());
            if (!$media = $cache->get($cacheKey)) {
                $media = new Media($this->getMediaFolder(), $this->getMediaOrder());
                $cache->set($cacheKey, $media);
            }
            $this->media = $media;
        }

        return $this->media;
    }

Turning off Grav cache and Twig cache don't help. We really appreciate if you could give any ideas to avoid this error. Ideally, if the requested page doesn't exist, Grav returns empty media list instead of throwing this error. Below is the stacktrace, maybe it is helpful.

```
TypeError thrown with message "Return value of Grav\Common\Page\Page::getCacheKey() must be of the type string, null returned"

Stacktrace:
#142 TypeError in /var/www/html/grav-test/system/src/Grav/Common/Page/Page.php:1205
#141 Grav\Common\Page\Page:getCacheKey in /var/www/html/grav-test/system/src/Grav/Common/Media/Traits/MediaTrait.php:74
#140 Grav\Common\Page\Page:getMedia in /var/www/html/grav-test/system/src/Grav/Common/Page/Page.php:1302
#139 Grav\Common\Page\Page:media in /var/www/html/grav-test/system/src/Grav/Common/Page/Page.php:654
#138 Grav\Common\Page\Page:content in /var/www/html/grav-test/system/src/Grav/Common/Page/Page.php:568
#137 Grav\Common\Page\Page:summary in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:702
#136 Twig\Template:getAttribute in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#135 __TwigTemplate_c5bfad1f5b13d0ae64f91548d1a3c2028baa168788a89445317e2c2c5ebb29d7:doDisplay in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:455
#134 Twig\Template:displayWithErrorHandling in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:422
#133 Twig\Template:display in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#132 __TwigTemplate_e29bdf3b0903e9273de7edeb2a175869ad6af82717ee8994bb62f3c73d9b0b6c:block_field in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:216
#131 Twig\Template:displayBlock in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#130 __TwigTemplate_e29bdf3b0903e9273de7edeb2a175869ad6af82717ee8994bb62f3c73d9b0b6c:doDisplay in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:455
#129 Twig\Template:displayWithErrorHandling in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:422
#128 Twig\Template:display in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#127 __TwigTemplate_5fbd4e72fab8a6fcbf94e5abbda3a5e6f28614b2fee45bab25f57c2f46ea6adf:block_field in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:216
#126 Twig\Template:displayBlock in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#125 __TwigTemplate_e0d1609d5c8fedf3a153065f6d2ab3ffefcbd0ebd67a191b16a7fb1b92120990:doDisplay in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:455
#124 Twig\Template:displayWithErrorHandling in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:422
#123 Twig\Template:display in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#122 __TwigTemplate_5fbd4e72fab8a6fcbf94e5abbda3a5e6f28614b2fee45bab25f57c2f46ea6adf:doDisplay in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:455
#121 Twig\Template:displayWithErrorHandling in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:422
#120 Twig\Template:display in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#119 __TwigTemplate_e29bdf3b0903e9273de7edeb2a175869ad6af82717ee8994bb62f3c73d9b0b6c:block_field in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:216
#118 Twig\Template:displayBlock in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#117 __TwigTemplate_e29bdf3b0903e9273de7edeb2a175869ad6af82717ee8994bb62f3c73d9b0b6c:doDisplay in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:455
#116 Twig\Template:displayWithErrorHandling in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:422
#115 Twig\Template:display in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#114 __TwigTemplate_0ce37ab8a25af81fb590b15cd564515377d0731ea6a7ae033770e4f02518c8f4___1760502572:doDisplay in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:455
#113 Twig\Template:displayWithErrorHandling in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:422
#112 Twig\Template:display in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#111 __TwigTemplate_0ce37ab8a25af81fb590b15cd564515377d0731ea6a7ae033770e4f02518c8f4:block_field in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:216
#110 Twig\Template:displayBlock in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#109 __TwigTemplate_e0d1609d5c8fedf3a153065f6d2ab3ffefcbd0ebd67a191b16a7fb1b92120990:doDisplay in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:455
#108 Twig\Template:displayWithErrorHandling in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:422
#107 Twig\Template:display in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#106 __TwigTemplate_0ce37ab8a25af81fb590b15cd564515377d0731ea6a7ae033770e4f02518c8f4:doDisplay in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:455
#105 Twig\Template:displayWithErrorHandling in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:422
#104 Twig\Template:display in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#103 __TwigTemplate_e29bdf3b0903e9273de7edeb2a175869ad6af82717ee8994bb62f3c73d9b0b6c:block_field in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:216
#102 Twig\Template:displayBlock in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#101 __TwigTemplate_e29bdf3b0903e9273de7edeb2a175869ad6af82717ee8994bb62f3c73d9b0b6c:doDisplay in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:455
#100 Twig\Template:displayWithErrorHandling in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:422
#99 Twig\Template:display in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#98 __TwigTemplate_5fbd4e72fab8a6fcbf94e5abbda3a5e6f28614b2fee45bab25f57c2f46ea6adf:block_field in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:216
#97 Twig\Template:displayBlock in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#96 __TwigTemplate_e0d1609d5c8fedf3a153065f6d2ab3ffefcbd0ebd67a191b16a7fb1b92120990:doDisplay in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:455
#95 Twig\Template:displayWithErrorHandling in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:422
#94 Twig\Template:display in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#93 __TwigTemplate_5fbd4e72fab8a6fcbf94e5abbda3a5e6f28614b2fee45bab25f57c2f46ea6adf:doDisplay in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:455
#92 Twig\Template:displayWithErrorHandling in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:422
#91 Twig\Template:display in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#90 __TwigTemplate_e29bdf3b0903e9273de7edeb2a175869ad6af82717ee8994bb62f3c73d9b0b6c:block_field in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:216
#89 Twig\Template:displayBlock in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#88 __TwigTemplate_e29bdf3b0903e9273de7edeb2a175869ad6af82717ee8994bb62f3c73d9b0b6c:doDisplay in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:455
#87 Twig\Template:displayWithErrorHandling in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:422
#86 Twig\Template:display in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#85 __TwigTemplate_0ce37ab8a25af81fb590b15cd564515377d0731ea6a7ae033770e4f02518c8f4___1760502572:doDisplay in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:455
#84 Twig\Template:displayWithErrorHandling in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:422
#83 Twig\Template:display in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#82 __TwigTemplate_0ce37ab8a25af81fb590b15cd564515377d0731ea6a7ae033770e4f02518c8f4:block_field in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:216
#81 Twig\Template:displayBlock in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#80 __TwigTemplate_e0d1609d5c8fedf3a153065f6d2ab3ffefcbd0ebd67a191b16a7fb1b92120990:doDisplay in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:455
#79 Twig\Template:displayWithErrorHandling in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:422
#78 Twig\Template:display in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#77 __TwigTemplate_0ce37ab8a25af81fb590b15cd564515377d0731ea6a7ae033770e4f02518c8f4:doDisplay in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:455
#76 Twig\Template:displayWithErrorHandling in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:422
#75 Twig\Template:display in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#74 __TwigTemplate_dca8c34dde63d14e746a1b3db3312e680417c85f9e1efdfbff91e2d5522eaafa:doDisplay in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:455
#73 Twig\Template:displayWithErrorHandling in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:422
#72 Twig\Template:display in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#71 __TwigTemplate_d5b22942e708f8107995da332d97e8fd8ec9972653efeee623905e91e0ba840f:block_content in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:216
#70 Twig\Template:displayBlock in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#69 __TwigTemplate_c35cce5fd54c843c97eceb28b15044b80df1b33389860618570e3e575569c490:block_content_wrapper in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:216
#68 Twig\Template:displayBlock in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#67 __TwigTemplate_c35cce5fd54c843c97eceb28b15044b80df1b33389860618570e3e575569c490:block_page in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:216
#66 Twig\Template:displayBlock in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#65 __TwigTemplate_c35cce5fd54c843c97eceb28b15044b80df1b33389860618570e3e575569c490:block_body in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:216
#64 Twig\Template:displayBlock in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#63 __TwigTemplate_c35cce5fd54c843c97eceb28b15044b80df1b33389860618570e3e575569c490:doDisplay in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:455
#62 Twig\Template:displayWithErrorHandling in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:422
#61 Twig\Template:display in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#60 __TwigTemplate_ebbd236609b7758107a69af700d26437a5f43b1c4cbcfcb108a57865ad0ab2fb:doDisplay in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:455
#59 Twig\Template:displayWithErrorHandling in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:422
#58 Twig\Template:display in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:497
#57 __TwigTemplate_d5b22942e708f8107995da332d97e8fd8ec9972653efeee623905e91e0ba840f:doDisplay in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:455
#56 Twig\Template:displayWithErrorHandling in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:422
#55 Twig\Template:display in /var/www/html/grav-test/vendor/twig/twig/src/Template.php:434
#54 Twig\Template:render in /var/www/html/grav-test/vendor/twig/twig/src/TemplateWrapper.php:47
#53 Twig\TemplateWrapper:render in /var/www/html/grav-test/vendor/twig/twig/src/Environment.php:384
#52 Twig\Environment:render in /var/www/html/grav-test/system/src/Grav/Common/Twig/Twig.php:388
#51 Grav\Common\Twig\Twig:processSite in /var/www/html/grav-test/system/src/Grav/Common/Service/OutputServiceProvider.php:28
#50 Grav\Common\Service\OutputServiceProvider:Grav\Common\Service\{closure} in /var/www/html/grav-test/vendor/pimple/pimple/src/Pimple/Container.php:118
#49 Pimple\Container:offsetGet in /var/www/html/grav-test/system/src/Grav/Common/Processors/RenderProcessor.php:28
#48 Grav\Common\Processors\RenderProcessor:process in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:45
#47 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:57
#46 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Common/Processors/DebuggerAssetsProcessor.php:28
#45 Grav\Common\Processors\DebuggerAssetsProcessor:process in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:45
#44 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:57
#43 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Common/Processors/PagesProcessor.php:69
#42 Grav\Common\Processors\PagesProcessor:process in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:45
#41 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:57
#40 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Common/Processors/TwigProcessor.php:27
#39 Grav\Common\Processors\TwigProcessor:process in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:45
#38 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:57
#37 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Common/Processors/AssetsProcessor.php:28
#36 Grav\Common\Processors\AssetsProcessor:process in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:45
#35 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:57
#34 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Common/Processors/SchedulerProcessor.php:29
#33 Grav\Common\Processors\SchedulerProcessor:process in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:45
#32 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:57
#31 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Common/Processors/BackupsProcessor.php:28
#30 Grav\Common\Processors\BackupsProcessor:process in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:45
#29 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:57
#28 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Common/Processors/TasksProcessor.php:59
#27 Grav\Common\Processors\TasksProcessor:process in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:45
#26 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:57
#25 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Common/Processors/RequestProcessor.php:52
#24 Grav\Common\Processors\RequestProcessor:process in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:45
#23 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:57
#22 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Common/Processors/ThemesProcessor.php:27
#21 Grav\Common\Processors\ThemesProcessor:process in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:45
#20 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:57
#19 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Common/Processors/PluginsProcessor.php:30
#18 Grav\Common\Processors\PluginsProcessor:process in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:45
#17 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:57
#16 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Common/Processors/InitializeProcessor.php:78
#15 Grav\Common\Processors\InitializeProcessor:process in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:45
#14 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:57
#13 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Common/Processors/DebuggerProcessor.php:27
#12 Grav\Common\Processors\DebuggerProcessor:process in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:45
#11 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:57
#10 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Common/Processors/ErrorsProcessor.php:27
#9 Grav\Common\Processors\ErrorsProcessor:process in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:45
#8 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:57
#7 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Common/Processors/LoggerProcessor.php:48
#6 Grav\Common\Processors\LoggerProcessor:process in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:45
#5 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:57
#4 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Common/Processors/ConfigurationProcessor.php:28
#3 Grav\Common\Processors\ConfigurationProcessor:process in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:45
#2 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php:57
#1 Grav\Framework\RequestHandler\RequestHandler:handle in /var/www/html/grav-test/system/src/Grav/Common/Grav.php:249
#0 Grav\Common\Grav:process in /var/www/html/grav-test/index.php:53

```

**Comment by mahagr** [comment]
Try setting page id, it drives the cache. Id needs to be unique in order to cache the rendered twig content.

**Comment by tranduyhung** [comment]
We have hundreds of page so it is almost impossible for us to set ID for all of them.

**Comment by mahagr** [comment]
See #2731

**Comment by mahagr** [comment]
Should be already fixed in Grav 1.6.18 (f59441eb5529732b55fdce1982cdd8a639f649ac)
